src/DependencyInjection/Providers/AzureProviderConfigurator.php

```diff
@@ -35,21 +39,26 @@ public function buildConfiguration(NodeBuilder $node)
             ->scalarNode('tenant')
                 ->example("tenant: 'common'")
                 ->info('The tenant to use, default is `common`')
+                ->defaultValue('common')
             ->end()
             ->scalarNode('url_api')
                 ->example("url_api: 'https://graph.windows.net/'")
                 ->info('Domain to build request URL')
+                ->defaultValue('https://graph.windows.net/')
             ->end()
             ->scalarNode('resource')
                 ->info('Oauth resource field')
+                ->defaultValue('')
```

Comment:
Is this required? I mean, is empty quotes different than allowing this just to be null?

Reply:
I'm just following what the Provider has set as their defaults:

```
class Azure extends AbstractProvider
{
    use BearerAuthorizationTrait;
    public $urlLogin = 'https://login.microsoftonline.com/';
    public $pathAuthorize = '/oauth2/authorize';
    public $pathToken = '/oauth2/token';
    public $scope = [];
    public $scopeSeparator = ' ';
    public $tenant = 'common';
    public $urlAPI = 'https://graph.windows.net/';
    public $resource = '';
    public $API_VERSION = '1.6';
    public $authWithResource = true;
...
}
```

I gave TheNetworg devs the benefit of the doubt. I figured they know what they're doing, so there may be a reason it needs to be an empty string as opposed to a null value.

Reply:
I can confirm that `null` value for resource works (I just went into my project and modified the config and the defaults in the class and authentication still works. I will update this to default to `null`.

Reply:
I updated the  `'resource'` node  with `->defaultNull()`